core/src/main/java/org/axonframework/deadline/quartz/DeadlineJob.java

```diff
@@ -0,0 +1,194 @@
+import org.axonframework.messaging.unitofwork.DefaultUnitOfWork;
+import org.axonframework.serialization.SerializedObject;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.serialization.SimpleSerializedObject;
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobDetail;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.SchedulerContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.axonframework.deadline.quartz.DeadlineJob.DeadlineJobDataBinder.deadlineMessage;
+import static org.axonframework.deadline.quartz.DeadlineJob.DeadlineJobDataBinder.deadlineScope;
+
+/**
+ * Quartz job which depicts handling of a scheduled deadline message. The {@link DeadlineMessage} and {@link
+ * ScopeDescriptor} are retrieved from the {@link JobExecutionContext}. The {@link TransactionManager} and
+ * {@link ScopeAware} components are fetched from {@link SchedulerContext}.
+ *
+ * @author Milan Savic
+ * @author Steven van Beelen
+ * @since 3.3
+ */
+public class DeadlineJob implements Job {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeadlineJob.class);
+
+    /**
+     * The key under which the {@link TransactionManager} is stored within the {@link SchedulerContext}.
+     */
+    public static final String TRANSACTION_MANAGER_KEY = TransactionManager.class.getName();
+
+    /**
+     * The key under which the {@link ScopeAwareProvider} is stored within the {@link SchedulerContext}.
+     */
+    public static final String SCOPE_AWARE_RESOLVER = ScopeAwareProvider.class.getName();
+
+    /**
+     * The key under which the {@link Serializer} is stored within the {@link SchedulerContext}.
+     */
+    public static final String JOB_DATA_SERIALIZER = Serializer.class.getName();
+
+    @Override
+    public void execute(JobExecutionContext context) throws JobExecutionException {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Starting a deadline job");
+        }
+
+        JobDetail jobDetail = context.getJobDetail();
+        JobDataMap jobData = jobDetail.getJobDataMap();
+
+        try {
+            SchedulerContext schedulerContext = context.getScheduler().getContext();
+
+            Serializer serializer = (Serializer) schedulerContext.get(JOB_DATA_SERIALIZER);
+            TransactionManager transactionManager = (TransactionManager) schedulerContext.get(TRANSACTION_MANAGER_KEY);
+            ScopeAwareProvider scopeAwareComponents = (ScopeAwareProvider) schedulerContext.get(SCOPE_AWARE_RESOLVER);
+
+            DeadlineMessage deadlineMessage = deadlineMessage(serializer, jobData);
+            ScopeDescriptor deadlineScope = deadlineScope(serializer, jobData);
+
+            DefaultUnitOfWork<DeadlineMessage<?>> unitOfWork = DefaultUnitOfWork.startAndGet(null);
+            Transaction transaction = transactionManager.startTransaction();
```

Comment:
I a problem arises starting this transaction, the `DefaultUnitOfWork` will not be properly closed.
It is safer to attach the transaction in the `execute` section.

See also: https://github.com/AxonFramework/AxonFramework/issues/616

Reply:
I have just pushed a commit to master (a398780), that addresses the Transaction start issue in a Unit of Work. There is now a method on the Unit of Work that can be used to attach a Transaction safely.